File: upa-impl-core/src/main/csharp/Sources/Custom/Util/ConvertedList_.cs
using System.Collections;
using System.Collections.Generic;

namespace Net.Vpc.Upa.Impl.Util
{


    /**
     * @author Taha BEN Salah
     * @creationdate 8/26/12 4:41 PM
     */
    public class ConvertedList<K, V> : AbstractReadOnlyList<V>
    {

        private Net.Vpc.Upa.Impl.Util.Converter<K , V> converter;

        private System.Collections.Generic.IList<K> @base;

        public ConvertedList(System.Collections.Generic.IList<K> @base, Net.Vpc.Upa.Impl.Util.Converter<K , V> converter) {
            this.@base = @base;
            this.converter = converter;
        }


        protected override V Get(int index)
        {
            return converter.Convert(@base[index]);
        }

        public override int Count
        {
            get { return @base.Count; }
        }
    }
}

File: upa-impl-core/src/main/java/net/vpc/upa/impl/util/ConvertedList.java
package net.vpc.upa.impl.util;

import net.vpc.upa.PortabilityHint;

import java.util.AbstractList;
import java.util.List;

/**
 * @author Taha BEN SALAH <taha.bensalah@gmail.com>
 * @creationdate 8/26/12 4:41 PM
 */
@PortabilityHint(target = "C#",name = "ignore")
public class ConvertedList<K,V> extends AbstractList<V>{
    private Converter<K,V> converter;
    private List<K> base;

    public ConvertedList(List<K> base, Converter<K, V> converter) {
        this.base = base;
        this.converter = converter;
    }

    @Override
    public V get(int index) {
        return converter.convert(base.get(index));
    }

    @Override
    public int size() {
        return base.size();
    }
}
